src/ui/views/session/entity-selected/EntitySelected.tsx

```diff
@@ -11,19 +11,20 @@ import { EntityData } from "../reticle/Reticle";
 import "./EntitySelected.css";
 
 export function EntitySelected({ entity }: { entity: EntityData | undefined }) {
-  const lastRef = useRef<EntityData>();
-  useEffect(() => {
-    lastRef.current = entity;
-  }, [entity]);
-
   const [isPeer, setIsPeer] = useState<boolean>();
+  const [isHeld, setIsHeld] = useState<boolean>(false);
 
   useEffect(() => {
     if (entity?.entityId) {
       const peer = entity?.peerId !== undefined;
       setIsPeer(peer);
+      if (entity?.held) {
+        setIsHeld(true);
+      } else {
+        setIsHeld(false);
+      }
     }
-  }, [entity, isPeer]);
+  }, [entity, isPeer, isHeld]);
```

Comment:
This seems like an anti-pattern and might cause an infinite loop since you're using `isHeld` as an input dependency.

Also:
```js
if (entity?.held) {
  setIsHeld(true);
} else {
  setIsHeld(false);
}
```

Can become `setIsHeld(entity?.held || false);`